src/transaction/log_recovery.c

```diff
@@ -2612,6 +2604,16 @@ log_recovery_analysis (THREAD_ENTRY * thread_p, LOG_LSA * start_lsa, LOG_LSA * s
 	  log_lsa.offset = lsa.offset;
 	  log_rec = LOG_GET_LOG_RECORD_HEADER (log_page_p, &log_lsa);
 
+	  if (is_media_crash == true)
+	    {
+	      /* Check also the last log page of current record. We need to check after obtaining log_rec. */
+	      is_log_page_broken = log_is_page_of_record_broken (thread_p, &log_lsa, log_rec);
+	      if (is_log_page_broken)
+		{
+		  break;
```

Comment:
please add comments to explain the logic. I was confused at the first sight. 
I think it will exit the inner loop and the log will be reset in the next iteration of the outer loop.